### DISCUSSION TITLE
Provide ID to resources

### DISCUSSION THREAD

**Discussion by asimpleidea** [CloudNativeSDWAN/cnwan-operator#56] [Ideas]
Different service registries have different method to store an object and index it.

For example:
* *Google Service Directory* indexes resemble a path: `locations/<location>/projects/<project-id>/namespace/<namespace/etc/<etc.>`
* *ETCD* indexes are similar to the above: `/prefix/namespaces/<namespace>/etc/<etc.>`
* DNS "indexes" (for lack of better word) can be `my-endpoint.my-service.my-namespace.cnwan.local`
* AWS are in the format of `aws:doeq9431ns` if I recall correctly.
* And so on...

While knowing the `ID` of the resource may not be useful for end users - i.e. reader or adaptor, they are necessary for other entities like the operator's *broker* because it needs to find them.

An update to the operator's service registry package will be brought which also includes an `ID string` field, which differs from the already existing `Name string` as the former will make sense for the specific service registry, while the latter for the user.
From the update on, `Name` will only contain the *user-friendly* name of the resource and what we are currently referring to as `Name` will be moved to `ID`.

---

Other names for `ID`:
* `ResourceName`

**Comment by arnatal** [comment]
Just to clarify @SunSince90, we're moving forward with this with ```Name``` and ```ResourceName```, is that correct?
